### PR TITLE
First pass at Freelancer Rates exercism

### PR DESCRIPTION
**Pull request by Theosaurus-Rex**
Heya, first time contributor to an Exercism track :)
I've done my best to adapt the Freelancer Rates exercise from the Elixir track to use Gleam.
Please let me know if there are any issues and I'll do my best to resolve them!

### CODE REVIEW

**Review by lpil — COMMENTED** [review]
Fantastic, thank you!

Would you like to also add a `.meta/example.gleam` example solution? https://github.com/exercism/gleam/blob/main/exercises/practice/hello-world/.meta/example.gleam

**Comment by Theosaurus-Rex** [comment]
> Fantastic, thank you!
> 
> Would you like to also add a `.meta/example.gleam` example solution? https://github.com/exercism/gleam/blob/main/exercises/practice/hello-world/.meta/example.gleam

Yep, sorry about that, knew I'd miss something! Have added an example solution :)

File: exercises/practice/freelancer-rates/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The returned monthly rate should be rounded up (take the ceiling) to the nearest integer.
+
+## 4. Calculate the number of workdays given a budget, hourly rate and discount
+
+Implement a function that takes a budget, a hourly rate, and a discount, and calculates how many days of work that covers.
+
+```
+freelancer_rates.days_in_budget(20000, 80, 11.0)
+# => 35.1
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.docs/instructions.md:52]: I believe that, given the implementation, but also the JavaScript version (I wrote this originally) that this should be rounded / floored! 

It also says it in the next line of the instructions.

```suggestion
# => 35
```

File: exercises/practice/freelancer-rates/.meta/config.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "blurb": "Using Floats and Integers, help a freelancer calculate their rates.",
+  "authors": [],
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.meta/config.json:3]: Add yourself! (GitHub username)

File: exercises/practice/freelancer-rates/.meta/config.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  "files": {
+    "solution": ["src/freelancer_rates.gleam"],
+    "test": ["test/freelancer_rates_test.gleam"],
+    "example": ["./example.gleam"]
+  }
+}
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.meta/config.json:9]: Missing a forked from value. If this was forked from elixir, that's what should be listed here.

```json
  "forked_from": ["elixir/freelancer-rates"],
```

File: exercises/practice/freelancer-rates/test/freelancer_rates_test.gleam [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+
+pub fn monthly_rate_test() {
+  freelancer_rates.monthly_rate(77, 10.5)
+  |> should.equal(12130.0)
+}
+
+pub fn days_in_budget_test() {
+  freelancer_rates.days_in_budget(20000, 80, 11.0)
+  |> should.equal(35.1)
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/test/freelancer_rates_test.gleam:28]:
```suggestion
  |> should.equal(35)
```

**Comment by SleeplessByte** [comment]
@Theosaurus-Rex since this track does not _yet_ run the example solution against the tests (it really should!) can you please run the tests against the example solution and ensure that it passes the tests? I have no clue if it does 😄

File: exercises/practice/freelancer-rates/.meta/config.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "blurb": "Using Floats and Integers, help a freelancer calculate their rates.",
+  "authors": [],
+  "files": {
+    "solution": ["src/freelancer_rates.gleam"],
+    "test": ["test/freelancer_rates_test.gleam"],
+    "example": ["./example.gleam"]
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.meta/config.json:7]:
```suggestion
    "example": [".meta/example.gleam"]
```

It's relative to the solution directory root!

File: exercises/practice/freelancer-rates/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+A month has 22 billable days.
+The freelancer is offering to apply a discount if the project manager chooses to let the freelancer bill per month, which can come in handy if there is a certain budget the project manager has to work with.
+
+Discounts are modeled as fractional numbers representing percentage, for example 25.0 (25%).
+
+## 1. Calculate the daily rate given an hourly rate
+
+Implement a function to calculate the daily rate given an hourly rate:
+
+```
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.docs/instructions.md:17]:
````suggestion
```gleam
````

File: exercises/practice/freelancer-rates/.docs/instructions.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+The returned daily rate should be a float.
+
+## 2. Calculate a discounted price
+
+Implement a function to calculate the price after a discount.
+
+```
+freelancer_rates.apply_discount(150, 10)
+# => 135.0
+```
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.docs/instructions.md:31]:
````suggestion
```gleam
freelancer_rates.apply_discount(150, 10)
# => 135.0
```
````

File: exercises/practice/freelancer-rates/.docs/instructions.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+The returned value should always be a float, not rounded in any way.
+
+## 3. Calculate the monthly rate, given an hourly rate and a discount
+
+Implement a function to calculate the monthly rate, and apply a discount:
+
+```
+freelancer_rates.monthly_rate(77, 10.5)
+# => 12130
+```
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.docs/instructions.md:42]:
````suggestion
```gleam
freelancer_rates.monthly_rate(77, 10.5)
# => 12130
```
````

**Comment by Theosaurus-Rex** [comment]
@SleeplessByte Thanks so much for the feedback! I've fixed up my example - still learning myself and forgot that floats have their own operators. @lpil please feel free to fix my solution, I feel like maybe I'm not handling floats VS integers as elegantly as I could

File: exercises/practice/freelancer-rates/.meta/example.gleam [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+pub fn monthly_rate(hourly_rate: Int, discount: Float) -> Int {
+  let before_discount = float.round(daily_rate(hourly_rate) *. 22.0)
+    let after_discount = apply_discount(before_discount, discount)
+    float.truncate(float.ceiling(after_discount))
+}
+
+pub fn days_in_budget(budget: Int, hourly_rate: Int, discount: Float) -> Float {
+     float.floor( int.to_float(budget) /. apply_discount(float.round(daily_rate(hourly_rate)), discount))
+}
```
Comment (SleeplessByte) [inline exercises/practice/freelancer-rates/.meta/example.gleam:21]:
```suggestion
pub fn days_in_budget(budget: Int, hourly_rate: Int, discount: Float) -> Int {
     float.truncate(int.to_float(budget) /. apply_discount(float.round(daily_rate(hourly_rate)), discount))
}
```

Perhaps like this? That way you could get the flooring + the conversion back to `Int` in one go.

Comment (SleeplessByte) [reply exercises/practice/freelancer-rates/.meta/example.gleam:21]: (If you accept this, don't forget to change the `stub` type signature.

**Review by SleeplessByte — COMMENTED** [review]
(I don't know Gleam, but I've followed Louis for years and see him talk about it on Twitter 😅)

File: exercises/practice/difference-of-squares/.meta/config.json [hunk 10]
```diff
@@ -1,20 +1,10 @@
...
-    ],
-    "test": [
-      "test/difference_of_squares_test.gleam"
-    ],
-    "example": [
-      ".meta/example.gleam"
-    ]
+    "solution": ["src/difference_of_squares.gleam"],
+    "test": ["test/difference_of_squares_test.gleam"],
+    "example": [".meta/example.gleam"]
```
Comment (SleeplessByte) [inline exercises/practice/difference-of-squares/.meta/config.json:7]: Did you mean to change this file?

File: exercises/practice/freelancer-rates/.meta/config.json (hunk 5 above)
Comment (SleeplessByte) [reply exercises/practice/freelancer-rates/.meta/config.json:7]: (You marked this as resolved but still there!)

**Review by lpil — APPROVED** [review]
Brilliant! Thank you!

**Pull request merged**